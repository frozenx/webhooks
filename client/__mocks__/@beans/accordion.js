

import React from 'react';

const Accordion = (props) => (<div {...props} />);

export default { Accordion };