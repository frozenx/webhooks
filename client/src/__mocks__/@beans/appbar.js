import React from 'react';
const AppBar = (props) => (<div {...props}/>);

export default AppBar;
