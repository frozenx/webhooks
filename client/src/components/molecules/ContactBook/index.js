import React, { Fragment } from "react";
import { FormHeading, PrimaryWrapper, SecondaryWrapper, FormHeadingWrapper } from "../../../common/styles";
import Button from '@beans/button';
import BackComponent from '../SupplierForm/BackComponent'
import CardListWithPagination from '../ContactCardList'
import i18Data from '../../../languagepack';

const ContactBook = ({
    showContactForm,
    contactList,
    getNewPage,
    partnerName,
    handleBackComponent,
    navigateToForm,
    showLoader,
    totalCount,
    currentPage,
    handlePageChange

}) => {
    return (
        <Fragment>
            <BackComponent handleBackComponent={handleBackComponent} />
            <PrimaryWrapper>
                <SecondaryWrapper>
                    <FormHeadingWrapper>
                        <FormHeading>
                            <h1>
                                {`${partnerName} ${i18Data.contacts}`}
                            </h1>
                        </FormHeading>
                        <Button onClick={showContactForm} variant='secondary'>
                            {i18Data.addNewContact}
                        </Button>
                    </FormHeadingWrapper>
                    <CardListWithPagination 
                        paginationProps = {
                            {
                                totalCount,
                                getNewPage, 
                                linkTemplate : "/page=%{pageNumber}",
                                pageText : "Page",
                                currentPage, 
                                handlePageChange
                            }
                        }
                        componentProps = {
                            {
                                contactList,
                                showLoader,
                                navigateToForm,
                                handleBackComponent

                            }
                        }
                    />
                </SecondaryWrapper>
            </PrimaryWrapper>
        </Fragment>
    );
}


export default ContactBook;