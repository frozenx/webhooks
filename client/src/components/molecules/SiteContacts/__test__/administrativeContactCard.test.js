import React from 'react';
import AdministrativeContactCard from '../AdminstrativeContactCard';
import renderer from 'react-test-renderer';
import { DefaultThemeProvider } from '@beans/theme';
import {shallow} from 'enzyme';
import ContactBook from '../../../pages/SupplierHeader/contactBook'


describe('SiteContacts', () => {
    let props,
        tree
    let contacts = [
        {
          "contactUuid": "774477537246487",
          "contactStatus": '',
          "firstName": "Volkswagen",
          "lastName": "Zetta",
        },
        {
          "contactUuid": "600853397549052",
          "contactStatus": "Active",
          "firstName": "Skoda",
          "lastName": "Rapid",          
        }
      ]    
   
    beforeEach(() => {
        props = {
            siteContactData: jest.fn(),
            editSiteContacts: jest.fn(),
            updateEditableStatus: jest.fn(),
            contacts: contacts,            
            
            node: {
                contact: {
                    values: {
                        "billToContact": "774477537246487",
                        "shipToContact": "",
                        "correspondenceContact": "",
                        "dunningContact": "",
                        "InvoicesContact": "",
                        "administrativeContact": "774477537246487"
                    }
                }
            },
                        
            rootNode: {
                uuid: 1234,
                contactBook: new ContactBook(contacts),                
                partner: {
                    values: {
                        partnerName: 'TEST NAME'                    
                    }
                }
            }

        }

        tree = renderer.create(
            <DefaultThemeProvider>
                <AdministrativeContactCard  {...props}/>
            </DefaultThemeProvider>
        ).toJSON();
    });
    test('should render the component', () => {
        expect(tree).toMatchSnapshot();
    });
    test('should call getContact method on componentDidMount', () => {
        const Wrapper = shallow(<AdministrativeContactCard {...props}/>);            
        const instance = Wrapper.instance();
        jest.spyOn(instance, 'getContact');
        instance.componentDidMount();
        expect(instance.getContact).toHaveBeenCalledWith(props.node.contact.values.administrativeContact);
    });
    test('should call the siteContactData of props method when changed or selected the addressType', () => {
        const mockEvent = { target: { value : '774477537246487'} };
        const Wrapper = shallow(<AdministrativeContactCard {...props}/>);
        Wrapper.find('.drop-down').first().simulate('change', mockEvent)
        expect(props.siteContactData).toHaveBeenCalled()
        expect(props.editSiteContacts).toHaveBeenCalledWith(false)
        expect(props.updateEditableStatus).toHaveBeenCalledWith(true)        
    }); 
    
    test('should not populate any values for contact types if there are no previously saved contactTypes', () => {
        const node = {
            contact: {
                values: {
                    "billToContact": "",
                    "shipToContact": "",
                    "correspondenceContact": "",
                    "dunningContact": "",
                    "InvoicesContact": "",
                    "administrativeContact": ""
                }
            }
        }
        const mockProps = {
            ...props, node
        }
        const Wrapper = shallow(<AdministrativeContactCard {...mockProps}/>);            
        expect(Wrapper.find('.contact-name')).toHaveLength(0);
    });

    test('should not return any contact if the contact is not found in the contactBook', () => {
        const node = {
            contact: {
                values: {
                    "billToContact": "",
                    "shipToContact": "",
                    "correspondenceContact": "",
                    "dunningContact": "",
                    "InvoicesContact": "",
                    "administrativeContact": "1234567"
                }
            }
        }
        const mockProps = {
            ...props, node            
        }
        const Wrapper = shallow(<AdministrativeContactCard {...mockProps}/>);            
        expect(Wrapper.find('.contact-name')).toHaveLength(0);
    });
    describe('when iseditSiteContactsEnable is true', () => {
        let Wrapper;
        beforeEach(() => {
            Wrapper = shallow(<AdministrativeContactCard {...props} iseditSiteContactsEnable={true}/>);
        })
        test('should show the close icon if the iseditSiteContactsEnable prop is true', () => {
            expect(Wrapper.find('.close')).toHaveLength(1)
        });
        test('should call siteContactData method of props if the close icon is clicked', () => {
            Wrapper.find('.close').simulate('click')
            expect(props.siteContactData).toHaveBeenCalled()
        });        
    })   
});