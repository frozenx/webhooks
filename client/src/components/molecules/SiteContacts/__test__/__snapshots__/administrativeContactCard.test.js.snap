// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteContacts should render the component 1`] = `
<div
  className="base-components__RootElement-sc-150tvch-1 sc-kAzzGY bphYLh"
>
  <div
    className="sc-kgoBCf ffTmed"
  />
  <div
    className="sc-kGXeez kScwyN"
  >
    <div
      className="sc-kpOJdX dPFWje"
    />
    <div
      className="sc-kpOJdX sc-dxgOiQ gHISXr"
    />
    <div
      className="sc-eNQAEJ kECEEu"
    >
      <div
        className="drop-down base-components__RootElement-sc-150tvch-1 cCmuza"
      >
        <select
          className="base-components__BaseInput-sc-150tvch-2 beans-dropdown__select styled__Select-sc-10jppqs-1 GLMMI"
          disabled={false}
          onChange={[Function]}
          role="listbox"
          value="select"
        >
          <option
            value="select"
          >
            Select contact
          </option>
          <option
            id="600853397549052"
            index={0}
            value="600853397549052"
          >
            Skoda,Rapid
          </option>
        </select>
        <svg
          aria-hidden={true}
          className="styled__AlignedIcon-sc-10jppqs-0 hHyXUl styled__StyledSVG-sc-1nulmu6-1 flhEYZ"
          color="currentColor"
          disabled={false}
          height={12}
          preserveAspectRatio="xMinYMax meet"
          viewBox="0 -1 20 12"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.652 1.312L10 10.276 1.356 1.312"
            fill="none"
            stroke="#666666"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
